File: microservices/services/transactions-service/app/schemas/transactions.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime

class CartItemAdd(BaseModel):
    course_id: int
    quantity: int = Field(1, ge=1)

class CartItemRemove(BaseModel):
    course_id: int

class CartItemOut(BaseModel):
    course_id: int
    course_title: str
    quantity: int
    unit_price: float
    subtotal: float

    class Config:
        from_attributes = True

class CartOut(BaseModel):
    count: int
    results: List[CartItemOut]
    total: float

class OrderOut(BaseModel):
    id: int
    number: str
    user_id: int
    status: str
    total: float

    class Config:
        from_attributes = True

class EnrollmentOut(BaseModel):
    id: int
    user_id: int
    course_id: int
    order_id: Optional[int] = None
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class CouponValidate(BaseModel):
    coupon_code: str

class CouponValidationResult(BaseModel):
    valid: bool
    code: str
    discount_percentage: float
    description: str

class QuoteItem(BaseModel):
    quantity: int = Field(..., ge=1)
    unit_price: float = Field(..., ge=0.0)

class QuoteRequest(BaseModel):
    items: List[QuoteItem]
    coupon_code: Optional[str] = None

class QuoteResponse(BaseModel):
    engine: str
    items_count: int
    subtotal: float
    discounts: float
    total: float
    currency: str

File: frontend/src/types/transactions.ts
export interface CartItem {
  course_id: number;
  course_title: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
}

export interface Cart {
  count: number;
  results: CartItem[];
  total: number;
}

export interface CartItemAddPayload {
  course_id: number;
  quantity: number;
}

export interface CouponValidatePayload {
  coupon_code: string;
}

export interface CouponValidationResult {
  valid: boolean;
  code: string;
  discount_percentage: number;
  description: string;
}

export interface Order {
  id: number;
  number: string;
  user_id: number;
  status: 'PENDING' | 'CONFIRMED' | 'CANCELLED';
  total: number;
}

export interface CheckoutReturnPayload {
  order_number: string;
  result: 'success' | 'fail' | 'cancel';
}

export interface Enrollment {
  id: number;
  user_id: number;
  course_id: number;
  order_id: number | null;
  status: string; // 'ACTIVA', etc.
  enrolled_at: string;
}
